### PR TITLE
new setup.sh

### PR DESCRIPTION
**Pull request by lzcalderaro**
New setup.

### CODE REVIEW

File: bin/setup.sh [hunk 1]
```diff
@@ -1,90 +1,94 @@
...
-docker-compose exec --user www-data phpfpm wp core update
-docker-compose exec --user www-data phpfpm wp core update-db
+wordpress_updater() {
+	## UPDATING COMPONENTS ##
+	echo "Updating WordPress"
+	docker-compose exec --user www-data phpfpm wp core update
+	docker-compose exec --user www-data phpfpm wp core update-db
+}
+
+read -p "Update Wordpress? (y/n)? default y: " is_multisite
```
Comment (Narayon) [inline bin/setup.sh]: I think WordPress should always be updated.

File: bin/setup.sh [hunk 2]
```diff
@@ -1,90 +1,94 @@
...
+		echo 'No config file found.'
+	fi
+}
+
+read -p "Wordpress repository, empty to clean wordpress: " REPOSITORY
 
 ## WORDPRESS SETUP ##
-if [ -f "./wordpress/wp-config.php" ]; then
-	echo "WordPress config file found."
+if [ ! -z "$REPOSITORY" ]; then
```
Comment (Narayon) [inline bin/setup.sh]: You need to check if a repo already exists, with ` && [ ! -d "./wordpress/.git" ]`

File: bin/setup.sh [hunk 3]
```diff
@@ -1,90 +1,94 @@
...
-docker-compose exec --user www-data phpfpm wp core update-db
+wordpress_updater() {
+	## UPDATING COMPONENTS ##
+	echo "Updating WordPress"
+	docker-compose exec --user www-data phpfpm wp core update
+	docker-compose exec --user www-data phpfpm wp core update-db
+}
+
+read -p "Update Wordpress? (y/n)? default y: " is_multisite
+case "$is_multisite" in
```
Comment (Narayon) [inline bin/setup.sh]: Aren't you missing the brackets - `{$is_multisite}`?

File: bin/setup.sh [hunk 4]
```diff
@@ -1,90 +1,94 @@
...
-if [ ! -z "$REPOSITORY" ] && [ ! -d "./wordpress/.git" ]; then
-	echo "No local repository found. Downloading boilerplate / project from GIT..."
-  cd "$ROOT/wordpress"
-	git clone $REPOSITORY .
-  cd "$ROOT"
+if ! echo "${site_url}" | grep '^[a-zA-Z0-9]*\.[a-zA-Z]*$' >/dev/null; then
+	URL="${URL}.local";
 fi
 
-## PROJECT DEPENDENCIES ##
```
Comment (Narayon) [inline bin/setup.sh]: Running `build.sh` is fundamental to this repo.

File: bin/setup.sh [hunk 5]
```diff
@@ -1,90 +1,94 @@
...
+	fi
+}
+
+read -p "Wordpress repository, empty to clean wordpress: " REPOSITORY
 
 ## WORDPRESS SETUP ##
-if [ -f "./wordpress/wp-config.php" ]; then
-	echo "WordPress config file found."
+if [ ! -z "$REPOSITORY" ]; then
+	echo "Local repository found. Downloading..."
```
Comment (Narayon) [inline bin/setup.sh]: It would be nice to print the name of the repo that is being cloned.

**Review by Narayon — CHANGES_REQUESTED** [review]

**Review by liciniofs — APPROVED** [review]

File: bin/setup.sh (hunk 1 above)
Comment (csrui) [reply bin/setup.sh]: Not if:

* You need to develop for a current installed version.
* Need to backtrack to test something.

**Review by csrui — APPROVED** [review]

File: bin/setup.sh (hunk 1 above)
Comment (Narayon) [reply bin/setup.sh]: Those use cases make sense, but that's a possibility not implemented in this setup.  
Normally, you don't have an installed version in the repo(WP installation is ignored) and you don't have a place to define what version should be installed.  
You can ask for the version, with the latest as default, and only update if no version is provided.  
Another possibility is to add all WP configuration to wp-cli.yml and let the repo decide what to do.  
I'm all in for the last option. 🙂

Comment (RicCastelhano) [reply bin/setup.sh]: When one needs to develop for a specific WP version, that should be typed by hand in the setup. This process should point us to the newest version possible. One should not start new projects already in legacy mode.

Comment (csrui) [reply bin/setup.sh]: Obviously one should not start a new project with an old version of WP @RicCastelhano. 

With this PR is possible @Narayon. As long as this script doesn't update "my" WP install, I can give this script the GIT url to my project and I can have a quick replica of my production, staging or whatever site or whatever version. If this script forces the update on me... well bollocks then. 

The update should be the developers choice give this script versatility.

Comment (Narayon) [reply bin/setup.sh]: For that to work, you need to have the WP installed on your repo. We don't do that and you shouldn't either.  
That's why I mentioned `wp-cli.yml` or `wp-config.php`.

File: bin/setup.sh [hunk 6]
```diff
@@ -1,90 +1,105 @@
...
+		break
+	else
+		echo "Please write a valid email."
+	fi
+
+done
+
+if [ ! -d "wordpress" ]; then
+  mkdir "wordpress"
+  chmod -R a+w "wordpress"
```
Comment (Narayon) [inline bin/setup.sh:37]: Isn't `wordpress` folder created by `docker-compose`?

**Review by Narayon — CHANGES_REQUESTED** [review]

File: bin/setup.sh (hunk 6 above)
Comment (lzcalderaro) [reply bin/setup.sh:37]: Nop, when i run setup, i got an error because the wordpress folder will not create.

File: bin/setup.sh (hunk 1 above)
Comment (csrui) [reply bin/setup.sh]: Right, I just got what you meant. 
No, we don't use WordPress it self registered on the repository. But let's leave that part aside for a bit as it is beside the point. 
Whats important here is that, no matter how a given WordPress version get's there, either composer, manual download or whatever, the developer should be able to keep that specific version and not be force fed the update.

Comment (Narayon) [reply bin/setup.sh]: I know what you want but I don't agree.  
First, this setup is not ready for what you want. How can I clone a repo and install a fresh WP?
Second, if we have a configuration file, why not use it?
Remember that this repo is specific to WP-Construct. We don't need to make it generic.

File: bin/setup.sh [hunk 7]
```diff
@@ -1,90 +1,93 @@
...
 fi
 
+read -p "Wordpress repository, empty to clean wordpress: " REPOSITORY
```
Comment (Narayon) [inline bin/setup.sh]: `Write your project repository URL (empty for a clean install):`

File: bin/setup.sh [hunk 8]
```diff
@@ -1,90 +1,93 @@
...
-	echo "WordPress config file found."
-else
-	echo "WordPress config file not found. Installing..."
+if [ ! -z "${REPOSITORY}" ]; then
+	
+	echo "Local repository found. Downloading..."
+	git clone $REPOSITORY "$ROOT/wordpress"
+fi
+
+if [ ! -f "$ROOT/wordpress/wp-config.php" ]; then
```
Comment (Narayon) [inline bin/setup.sh]: What if I have a wp-config.php on my repo, but not a WP installation (this is my current project use case)?

**Review by Narayon — CHANGES_REQUESTED** [review]

File: bin/setup.sh [hunk 9]
```diff
@@ -1,90 +1,104 @@
...
+esac
+
+wordpress_updater() {
+	## UPDATING COMPONENTS ##
+	echo "Updating WordPress"
+	docker-compose exec --user www-data phpfpm wp core update
+	docker-compose exec --user www-data phpfpm wp core update-db
+}
+
+if [ "${is_install}" == 'n' ] || [ "${is_install}" == "N" ]; then
```
Comment (Narayon) [inline bin/setup.sh]: You don't need this validation.  
Even if you install WP core, you can always ask for an update.

**Review by Narayon — CHANGES_REQUESTED** [review]

**Review by Narayon — APPROVED** [review]

**Pull request merged**